src/Workspaces/Core/Portable/Experiments/IExperimentationService.cs

```diff
@@ -1,27 +1,35 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
 
 namespace Microsoft.CodeAnalysis.Experiments
 {
-    internal interface IExperimentationService : IWorkspaceService
+    internal interface IExperimentationServiceFactory : IWorkspaceService
     {
-        bool IsExperimentEnabled(string experimentName);
+        ValueTask<IExperimentationService> GetExperimentationServiceAsync(CancellationToken cancellationToken);
```

Comment:
Comments please as to why this is now split up.

Reply:
The separation falls out from the default explanation for all MEF services:

* The constructor for `IExperimentationServiceFactory` cannot call asynchronous code
* The `IsExperimentEnabled` method needs to be callable from synchronous code paths
* The `IsExperimentEnabled` method has a precondition that an asynchronous code path was called

None of these items seems especially worthwhile to put in code for _this service_, but if we have documentation somewhere for MEF practices we could mention it.